diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SmartCastRule.kt

```diff
@@ -0,0 +1,286 @@
+            if (it.node.treeParent.text.contains(" is ")) {
+                groups.getValue(it).forEach { asCall ->
+                    if (asCall.node.findParentNodeWithSpecificType(THEN) != null) {
+                        raiseWarning(asCall.node)
+                    }
+                }
+            } else if (it.node.treeParent.text.contains(" !is ")) {
+                groups.getValue(it).forEach { asCall ->
+                    if (asCall.node.findParentNodeWithSpecificType(ELSE) != null) {
+                        raiseWarning(asCall.node)
+                    }
+                }
+            }
+        }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun raiseWarning(asCall: ASTNode) {
+        SMART_CAST_NEEDED.warnAndFix(configRules, emitWarn, isFixMode, asCall.treeParent.text, asCall.startOffset,
+                asCall) {
+            val dotExpr = asCall.findParentNodeWithSpecificType(DOT_QUALIFIED_EXPRESSION)!!
+            val afterDotPart = dotExpr.text.split(".")[1]
+            val text = "${asCall.text}.$afterDotPart"
+            dotExpr.treeParent.addChild(KotlinParser().createNode(text), dotExpr)
+            dotExpr.treeParent.removeChild(dotExpr)
+        }
+    }
+
+    /**
+     * Groups is and as expressions, so that their are used in same if block
+     */
+    private fun groupIsAndAsExpr(isExpr: List<KtNameReferenceExpression>,
+                                 asExpr: List<KtNameReferenceExpression>,
+                                 prop: KtProperty)
+            : Map<KtNameReferenceExpression, List<KtNameReferenceExpression>> {
+        val groupedExprs = mutableMapOf<KtNameReferenceExpression, List<KtNameReferenceExpression>>()
+
+        if (isExpr.isEmpty() && asExpr.isNotEmpty()) {
+            handleZeroIsCase(asExpr, prop)
+            return emptyMap()
+        }
+
+        isExpr.forEach {
+            val list = mutableListOf<KtNameReferenceExpression>()
+            asExpr.forEach { asCall ->
+                if (asCall.node.findParentNodeWithSpecificType(IF)
+                        == it.node.findParentNodeWithSpecificType(IF))
+                    list.add(asCall)
+            }
+            groupedExprs[it] = list
+        }
+        return groupedExprs
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun getPropertyType(prop: KtProperty): String {
+        when (prop.initializer?.node?.elementType) {
+            STRING_TEMPLATE -> return "String"
+            INTEGER_CONSTANT -> return "Int"
+        }
+        if (prop.initializer?.node?.elementType == FLOAT_CONSTANT) {
+            if (prop.initializer?.text!!.endsWith("f"))
+                return "Float"
+            return "Double"
+        }
+        return "Not defined"
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun handleZeroIsCase(asExpr: List<KtNameReferenceExpression>, prop: KtProperty) {
+        val propType = getPropertyType(prop)
+        if (propType == "Not defined")
```

Comment:
In which case is this happening? It'll be better if you return nullable string from `getPropertyType` and then do `?: return`